src/lib.rs

```diff
@@ -525,6 +520,34 @@ impl<'a> AsciiStlReader<'a> {
     }
 }
 
+fn tri_area(a: Vertex, b: Vertex, c: Vertex) -> f32 {
```

Comment:
What is the advantage of using macros here? Couldn't we define vector_cross, vector_sub and vector_length as normal functions?

Reply:
We certainly could use functions. One advantage that macro's might have is that the the binary program would be (slightly) improved as the macro code will be inlined. However I believe that the rust compiler will apply these improvements for functions as well. I think it just comes down to personal preference.. before i wrote the code i was reading about macros, thats probably why i used them. Could easily change to functions if you'd like

Also one question, would you rather that these function would be defined inside 'tri_area' or would you rather have them defined in the modules' scope?

Reply:
I doubt one can gain measurable performance benefits using macros in this case.
So let's use functions.
As for scope: I'm fine either way.

Reply:
all done!

Reply:
Thanks a lot!